### DISCUSSION TITLE
FYI, there is a critical low-level bug fixed in Texas Instruments Z-Stack 4.40.00.44 SDK

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#655] [General]
FYI, Wwalltt (@walthowd?) posted this TI 4.40.00.44 + firmware upgrade recommendation yesterday on Home Assistant subreddit on Reddit:

https://www.reddit.com/r/homeassistant/comments/l9uhvm/psa_zigbee_network_lockup_bug_affecting/

### PSA: Zigbee network lockup bug affecting Zigbee2mqtt and ZHA (zigpy-znp) fixed

Just as a public service announcement, a long-standing bug in Texas Instruments Z-Stack that caused lockups and network shutdown on zigbee2mqtt and ZHA (when using a TI board) has just been fixed.

The bug would be very rare for some or most users, but more common for users with a large (100+) number of devices. The symptoms would be loss of communication with the Zigbee coordinator radio. The serial port would still be present, but nothing would be returned from Z-Stack.

This bug affected:

- Electrolama Zig-a-Zig-Ah (ZZH)
- Slaesh CC2652 stick
- Texas Instruments CC1352P2
- Texas Instruments CC26X2R1
- Texas Instruments CC2538

Here is the link to the updated ZStack firmware (dated 2021-01-20) that is supposed to fix the issue:

https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin

**Update**! Link updated as Koenkk has moved these updated firmware image files from the "develop" tree to the "master" (mainline) tree.

Here are the Texas Instrument release notes:

http://software-dl.ti.com/simplelink/esd/simplelink_cc13x2_26x2_sdk/4.40.00.44/exports/changelog.html

ZIGBEE-1172 NPI memory leak from buffer deallocation order

Update today and share your results if this issue has been affecting you.

**Note!** Highly recommend that you backup NVRAM of Texas Instruments adapters with zigpy-znp before upgrading firmware:

https://github.com/zigpy/zigpy-znp#nvram

**Tip!** If you do a backup of NVRAM before then can do a restore of NVRAM after if have any issues or need to downgrade later.

**Comment by Hedda** [comment]
Not sure if this affects CC2530/CC2531 **with TI Z-Stack 3 firmware** in practice as will probably not have 100+ devices on those.

**Reply by MattWestb** [reply]
One thing the newest FW is 20210120 so they is not fixed or is the date wrong.
 (dated 1/25/21) = 20210125 if i counting right ;-)

**Comment by Hedda** [comment]
FYI, zigpy-znp developer puddly posted in https://github.com/zigpy/zigpy-znp/issues/62 recommending upgrade to this firmware on all CC2652 and CC1352

https://github.com/zigpy/zigpy-znp/issues/62

> Yes, it is very much recommended. I have been reliably able to trigger Z-Stack crashes on older firmwares in about 3-4 hours (by sending requests as fast as possible) that are impossible to reset remotely on both the LAUNCHXL boards and on the ZZH. Slaesh's stick should in theory be able to recover from any lockup with the unreleased reconnect watchdog in the `dev` branch. New firmware has been running for over 36 hours now without issues.

Other people replied to https://www.reddit.com/r/homeassistant/comments/l9uhvm/psa_zigbee_network_lockup_bug_affecting/

**Comment by Hedda** [comment]
By the way, Koenkk has also moved the updated firmware image files from the "develop" tree to the "master" (mainline) tree:

https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin
